script/hassfest/manifest.py

```diff
@@ -366,15 +366,19 @@ def _sort_manifest_keys(key: str) -> str:
     return _SORT_KEYS.get(key, key)
 
 
-def sort_manifest(integration: Integration) -> bool:
+def sort_manifest(integration: Integration, config: Config) -> bool:
     """Sort manifest."""
     keys = list(integration.manifest.keys())
     if (keys_sorted := sorted(keys, key=_sort_manifest_keys)) != keys:
         manifest = {key: integration.manifest[key] for key in keys_sorted}
         integration.manifest_path.write_text(json.dumps(manifest, indent=2))
```

Comment:
Should these lines be moved to under `if config.action == "generate":` as well?
There is not need to write this unless we are generating it?